core/node/node_framework/src/implementations/layers/base_token_fetcher.rs

```diff
@@ -0,0 +1,72 @@
+use std::sync::Arc;
+
+use zksync_base_token_adjuster::BaseTokenFetcher;
+use zksync_config::BaseTokenAdjusterConfig;
+
+use crate::{
+    implementations::resources::{
+        base_token_fetcher::BaseTokenFetcherResource,
+        pools::{PoolResource, ReplicaPool},
+    },
+    service::{ServiceContext, StopReceiver},
+    task::{Task, TaskId},
+    wiring_layer::{WiringError, WiringLayer},
+};
+
+/// Wiring layer for `BaseTokenFetcher`
+///
+/// Responsible for serving the latest ETH<->BaseToken conversion ratio. This layer is only wired if
+/// the base token is not ETH. If wired, this layer inserts the BaseTokenFetcherResource and kicks
+/// off a task to poll the DB for the latest ratio and cache it.
+///
+/// If the base token is ETH, a default, no-op impl of the BaseTokenFetcherResource is used by other
+/// layers to always return a conversion ratio of 1.
+
+/// ## Requests resources
+///
+/// - `PoolResource<MasterPool>`
+///
+/// ## Adds tasks
+///
+/// - `BaseTokenFetcher`
+#[derive(Debug)]
+pub struct BaseTokenFetcherLayer {
+    config: BaseTokenAdjusterConfig,
+}
+
+impl BaseTokenFetcherLayer {
+    pub fn new(config: BaseTokenAdjusterConfig) -> Self {
+        Self { config }
+    }
+}
+
+#[async_trait::async_trait]
+impl WiringLayer for BaseTokenFetcherLayer {
+    fn layer_name(&self) -> &'static str {
+        "base_token_fetcher"
+    }
+
+    async fn wire(self: Box<Self>, mut context: ServiceContext<'_>) -> Result<(), WiringError> {
+        let replica_pool_resource = context.get_resource::<PoolResource<ReplicaPool>>().await?;
+        let replica_pool = replica_pool_resource.get().await.unwrap();
+
+        let fetcher = BaseTokenFetcher::new(Some(replica_pool), self.config);
+
+        context.insert_resource(BaseTokenFetcherResource(Arc::new(fetcher.clone())))?;
+        context.add_task(Box::new(fetcher));
+
+        Ok(())
+    }
+}
+
+#[async_trait::async_trait]
+impl Task for BaseTokenFetcher {
+    fn id(&self) -> TaskId {
+        "base_token_fetcher".into()
+    }
+
+    async fn run(mut self: Box<Self>, stop_receiver: StopReceiver) -> anyhow::Result<()> {
+        (*self).run(stop_receiver.0).await?;
+        Ok(())
+    }
```

Comment:
Now that I thought about it more -- it looks like the task approach won't work.
GasAdjuster task [has a check](https://github.com/matter-labs/zksync-era/blob/main/core/node/node_framework/src/implementations/layers/l1_gas.rs#L103) that only actually runs the task if someone uses the adjuster. However, here we can't do the same -- because the only user of the resource is the `GasAdjuster`, and it always has a reference to the base token fetcher.

Probably we have no choice but to add an `update` method to the `BaseTokenFetcher`, and invoke it from `GasAdjuster::keep_updated`

Reply:
Leaving this as is as discussed offline.